models/Asset.php

```diff
@@ -966,77 +966,106 @@ public function delete(bool $isNested = false): void
-                new VersionDeleteMessage(Service::getElementType($this), $this->getId())
-            );
+                try {
+                    $this->closeStream();
+
+                    // remove children
+                    if ($this->hasChildren()) {
+                        foreach ($this->getChildren() as $child) {
+                            $child->delete(true);
+                            $deletedPaths[] = $child->getRealPath() . $child->getId();
+                            $deletedIds[] = $child->getId();
+                        }
+                    }
 
-            // remove all properties
-            $this->getDao()->deleteAllProperties();
+                    // Dispatch Symfony Message Bus to delete versions
+                    Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
+                        new VersionDeleteMessage(Service::getElementType($this), $this->getId())
+                    );
 
-            // remove all tasks
-            $this->getDao()->deleteAllTasks();
+                    // remove all properties
+                    $this->getDao()->deleteAllProperties();
 
-            // remove dependencies
-            $d = $this->getDependencies();
-            $d->cleanAllForElement($this);
+                    // remove all tasks
+                    $this->getDao()->deleteAllTasks();
 
-            // remove from resource
-            $this->getDao()->delete();
+                    // remove dependencies
+                    $d = $this->getDependencies();
+                    $d->cleanAllForElement($this);
 
-            $this->commit();
+                    // remove from resource
+                    $this->getDao()->delete();
 
-            // remove file on filesystem
-            if (!$isNested) {
-                $fullPath = $this->getRealFullPath();
-                if ($fullPath != '/..' && !strpos($fullPath,
-                    '/../') && $this->getKey() !== '.' && $this->getKey() !== '..') {
-                    $this->deletePhysicalFile();
-                }
+                    $this->commit();
+
+                    // remove file on filesystem
+                    if (!$isNested) {
+                        $deletedPaths[] = $this->getRealPath() . $this->getId();
+                        $deletedIds[] = $this->getId();
+
+                        $fullPath = $this->getRealFullPath();
+                        if ($fullPath != '/..' && !strpos($fullPath,
+                                '/../') && $this->getKey() !== '.' && $this->getKey() !== '..') {
+                            $this->deletePhysicalFile();
+                        }
+
+                        $this->clearThumbnailsByPaths($deletedPaths);
+                        $this->getDao()->deleteFromThumbnailCacheByIds($deletedIds);
+                        //remove target parent folder preview thumbnails
+                        $this->clearFolderThumbnails($this);
```

Comment:
moved the thumbnail clear to the end of the main `delete()` but could have problem similar to what we encountered in https://github.com/pimcore/pimcore/issues/17534